src/handlers/cloudtrail.py

```diff
@@ -0,0 +1,47 @@
+import tornado.web
+import logging
+import sys
+import gzip
+import json
+
+from src.lib.Statsd import StatsClientSingleton
+
+
+class CloudTrailHandler(tornado.web.RequestHandler):
+    """ The AWS CloudTrail handler class
+    """
+
+    def initialize(self, amqp_con):
+        """
+        handler initialisation
+        """
+        self.logger = logging.getLogger("tornado.application")
+        self.amqp_con = amqp_con
+
+    def post(self):
+        """
+        HTTP Post handler
+        * Forward request: publish to AMQP
+        :return: HTTPStatus 200
+        """
+        try:
+            StatsClientSingleton().incr('input.cloudtrail', count=1)
+            StatsClientSingleton().incr('amqp.output', count=1)
+            routing_key = self.request.uri.replace('/', '.')[1:]
+
+            payload = self.request.body
+            content_encoding = self.request.headers.get('Accept-Encoding')
+            if content_encoding == 'gzip':
+                payload = gzip.decompress(payload)
+
+            entry_list = json.loads(payload)["Records"]
```

Comment:
```json.loads(payload).get("Records", [])``` will fallback to an empty array if the field "Records" is missing

Reply:
That's not supposed to happen, so it seems relevant to let the error happen return a 500 error code (which is currently the case). Should I do this change and manually return 500 ?